### ISSUE TITLE
Better CLI login

### ISSUE THREAD

**Issue by kontsevoy** [gravitational/teleport#665]
As Teleport gains more authentication options, `tsh` needs to provide a better experience for users who opt for non-default options (and work out of the box without additional flags), i.e. `tsh login` or `tsh ssh` should work equally well for:

* Auth0 users
* U2F users

**Comment by klizhentas** [comment]
I need this feature to unblock telekube logins that need second factor to be turned off. Here's my proposal:

## New resource type

Introduce new type of resource `ClusterAuthPreference`

* Local authentication without second factor:

```yaml
kind: ClusterAuthPreference
spec:
   auth_type: "local"
   params:
       second_factor:
            enabled: false
```


* Local authentication with U2F factor

```yaml
kind: ClusterAuthPreference
spec:
   auth_type: "local"
   params:
       second_factor:
            enabled: true
            type: u2f
```

* External authentication with OIDC google

```yaml
kind: ClusterAuthPreference
spec:
   auth_type: "oidc"
   params:
       connector_id: "google"
```

## New auth server API method

`GET /v2/cluster-auth-preferences`

is method that does not need authentication and returns the resource value.

tsh client and web proxy can simply read this method to get the values and render the UI form and set command line flags.

**Comment by klizhentas** [comment]
thinking about this later, we don't need this to be a resource, we can simply make this a static configuration section. We would still need to expose this over the web API though, so Auth server has to expose the API over this to communicate to web proxies.

**Comment by klizhentas** [comment]
actually, take it back - resource is probably better and gives us this simple way to reason about all our configs

* we have a set of resources: users, roles and others, like `ClusterAuthPreference`
* we can have them static and supply via config with `seed_config: true`. 

In this case, everything is read only. we can have them dynamic, and simply configure everything via API

so people using ansible can drop off their resources `/var/lib/teleport/conf.d` and send `HUP`

and others using teleport as a library will use API to build UI, settings, etc

**Comment by kontsevoy** [comment]
duplicate of #758
